modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsEntityByChart.php

```diff
@@ -141,4 +123,47 @@ public function blockSubmit($form, FormStateInterface $form_state) {
     $this->configuration['entity_by'] = $form_state->getValue('entity_by');
   }
 
+  /**
+   * Get data for chart.
+   *
+   * @return array
+   *   Data.
+   */
+  private function getData() {
+    $entity_by = $this->configuration['entity_by'];
+    $storage = $this->entityTypeManager->getStorage('node');
+    $entity_query = $storage->getQuery();
+    $query = $entity_query->condition('type', 'idea');
+    $filters = $this->getFilters();
+    if (isset($filters[OpenidealStatisticsDateSelectForm::TO])) {
+      $query->condition('created', $filters[OpenidealStatisticsDateSelectForm::TO], '<=');
+    }
+    elseif (isset($filters[OpenidealStatisticsDateSelectForm::FROM])) {
+      $query->condition('created', $filters[OpenidealStatisticsDateSelectForm::FROM], '>=');
+    }
+
+    if ($entity_by == 'category') {
+      $query->exists('field_category');
+    }
+
+    $ids = $query->execute();
+
+    $ideas = $storage->loadMultiple($ids);
```

Comment:
I think we may have different entity types and bundles here (not only ideas).

Reply:
Only Ideas were requested.